### PR TITLE
Server updates 0.4.11

### PR DESCRIPTION
**Pull request by athola**
## Summary

This PR delivers two focused improvements to the skrills codebase:

1. **Test Infrastructure Refactoring** - Converts manual environment variable cleanup patterns to RAII guards across `server` and `intelligence` crates, preventing race conditions in parallel tests
2. **Documentation Consolidation** - Converts bulleted lists to flowing prose across 14 documentation files for improved readability

### Test Isolation Improvements

- Add `test_support` module to `intelligence` crate with `env_guard()` and `set_env_var()` helpers
- Add `TestFixture` struct to `server` crate for standardized test setup
- Convert 16 tests from manual cleanup to `EnvVarGuard` RAII pattern
- Fix race condition causing intermittent test failures

### New Test Coverage

- 5 tests for `compute_feature_distance` edge cases
- 2 tests for cluster formation behavior (session clustering, label assignment)
- 6 tests for deviation score boundary conditions
- 5 tests for n-gram extraction edge cases
- 7 tests for `build_behavioral_patterns`
- 1 test for `parse_trace_target` case sensitivity
- 1 test for manifest save on read-only directories

### Documentation

- Convert bulleted lists to prose in README, book chapters, and technical docs
- Net reduction of 65 lines while preserving all technical content
- Improve API key error messages with helpful context and links

### Code Quality

- Replace `.expect()` with defensive `if-let` pattern in `tools.rs`
- Add doc comments to tool functions
- Add debug/trace logging for edge cases

## Related Issues

Addresses issues #83 and #86 from PR triage.

## Test Plan

See test plan comment below.

### CODE REVIEW

**Comment by athola** [comment]
## PR Review: Server Updates 0.4.11

### Summary

This PR delivers two well-scoped improvements:

1. **Test Infrastructure Refactoring** - Converts manual env var cleanup to RAII guards across `server` and `intelligence` crates
2. **Documentation Consolidation** - Converts bulleted lists to flowing prose across 14 documentation files

### Scope Assessment ✅

Both commits are tightly focused:
- **Commit 1**: Test isolation improvements (addresses issues #83 and #86)
- **Commit 2**: Documentation readability + 2 cluster formation tests

No scope creep detected. Changes are incremental and well-justified.

### Code Quality Analysis

#### Strengths

1. **RAII Pattern Implementation** - The `EnvVarGuard` and `TestFixture` abstractions in both crates are well-designed:
   - Automatic cleanup prevents env var leaks between parallel tests
   - `env_guard()` mutex serializes tests that mutate process-global state
   - `TestFixture::new()` standardizes test setup with common directory structures

2. **Defensive Coding** - Good improvement in `crates/server/src/app/tools.rs:184`:
   ```rust
   // Changed from .expect() to if-let for defensive pattern
   if let Some(skill_object) = skill_json.as_object_mut() { ... }
   ```

3. **Improved Error Messages** - API key error messages now include helpful context:
   ```rust
   "Claude API key not set. Set SKRILLS_CLAUDE_API_KEY... Get one at https://console.anthropic.com/settings/keys"
   ```

4. **New Test Coverage** - Added tests for:
   - `compute_feature_distance` (5 test cases)
   - `cluster_sessions` cluster formation (2 test cases)
   - Deviation score boundary conditions (6 test cases)
   - N-gram edge cases (5 test cases)
   - `build_behavioral_patterns` (7 test cases)
   - Case sensitivity in `parse_trace_target`
   - Manifest save on read-only directories

5. **Documentation Quality** - Prose consolidation reduces visual noise while preserving all technical content. Net reduction of 65 lines.

#### Minor Observations (Non-Blocking)

1. **Debug logging added** at `crates/intelligence/src/create/empirical.rs:665`:
   ```rust
   debug!(cluster_id = %cluster.cluster_id, "no success patterns for confidence, defaulting to 0.5");
   ```
   Good for observability - no action needed.

2. **Trace logging added** at `crates/intelligence/src/usage/behavioral.rs:195`:
   ```rust
   trace!("timestamp missing or invalid in tool_use block, defaulting to 0");
   ```
   Appropriate log level for this edge case.

3. **Doc comments added** to tool functions in `tools.rs` - These improve API discoverability.

### Version Consistency ✅

All crates remain at version `0.4.10`. The CHANGELOG correctly places these changes under "Unreleased" section.

### Security Review ✅

- No new attack surface introduced
- The read-only directory test (`test_manifest_save_fails_on_readonly_parent`) properly restores permissions on cleanup
- No secrets or credentials in changes

### Test Results ✅

```
All tests pass
Clippy: No warnings
```

### Verdict: **APPROVE**

Well-executed maintenance PR that improves test reliability and documentation readability. The RAII patterns will prevent flaky tests from env var race conditions.

File: crates/subagents/src/backend/claude.rs [hunk 1]
```diff
@@ -69,17 +69,18 @@ impl ClaudeAdapter {
             .await?;
 
         if self.config.api_key.is_empty() {
+            let error_msg = "Claude API key not set. Set SKRILLS_CLAUDE_API_KEY environment variable with your Anthropic API key. Get one at https://console.anthropic.com/settings/keys";
```
Comment (athola) [inline crates/subagents/src/backend/claude.rs]: this should be a const and set at the top of the page

File: crates/subagents/src/backend/codex.rs [hunk 2]
```diff
@@ -67,17 +67,18 @@ impl CodexAdapter {
             .await?;
 
         if self.config.api_key.is_empty() {
+            let error_msg = "Codex API key not set. Set SKRILLS_CODEX_API_KEY environment variable with your OpenAI API key. Get one at https://platform.openai.com/api-keys";
```
[the remark this answers is not in the document]
Comment (athola) [reply crates/subagents/src/backend/codex.rs]: Fixed - extracted to `API_KEY_ERROR` constant at the top of the file in commit 38bc886.
